internal/api/client.go

```diff
@@ -102,9 +120,11 @@ func (d *DetaClient) request(i *requestInput) (*requestOutput, error) {
 				return nil, fmt.Errorf("failed to get access token: %w", err)
 			}
 		}
-		//  request timestamp
-		now := time.Now().UTC().Unix()
-		timestamp := strconv.FormatInt(now, 10)
+
+		timestamp, err := fetchServerTimestamp()
```

Comment:
When running most commands we are making multiple authenticated requests and with this we would be making one additional request for each of them if I understand it correctly. This does not seem very efficient and might actually slow things down.

What is the issue with using the local timestamp? Can we not just convert the local timestamp into the server's timezone instead?

If not, is there a way to reduce it so we only have to fetch the server timestamp once and then use it to calculate the right timestamp for the subsequent requests?

Reply:
Basically the issue is: you cannot trust client timestamps, especially on windows. There are routinely shifted from several minutes, and we got a lot of errors regarding this issue. It is not a timezone issue, just a system clock problem.

> If not, is there a way to reduce it so we only have to fetch the server timestamp once and then use it to calculate the right timestamp for the subsequent requests?

This is a nice suggestion, I'll implement this